vowpalwabbit/options.h

```diff
@@ -53,9 +56,34 @@ struct option_builder
     return *this;
   }
 
+  template <typename U>
+  void add_help(std::ostringstream&, const U&)
+  {
+    THROW("Error: cannot handle non-string or arithmetic types in one_of().");
+  }
+  void add_help(std::ostringstream& help, const std::string& addition) { help << addition; }
+  void add_help(std::ostringstream& help, const int& addition) { help << std::to_string(addition); }
+
   option_builder& help(const std::string& help)
   {
-    m_option_obj.m_help = help;
+    std::ostringstream help_w_additions;
+    help_w_additions << help;
+    if (m_option_obj.one_of().size() > 0)
+    {
+      help_w_additions << ". Choices: {";
+      bool first = true;
+      for (const auto& v : m_option_obj.one_of())
+      {
+        if (first) { first = false; }
+        else
+        {
+          help_w_additions << ", ";
+        }
+        add_help(help_w_additions, v);
+      }
+      help_w_additions << "}";
+    }
+    m_option_obj.m_help = help_w_additions.str();

```

Comment:
This could be replaced with something like:

```c++
m_option_obj.m_help = fmt::format("{}. Choices {{{}}}", fmt::join(m_option_obj.one_of(), ", ");
```